pkg/apis/projectcalico/v3/ippool.go

```diff
@@ -78,8 +78,27 @@ type IPPoolSpec struct {
 	// Deprecated: this field is only used for APIv1 backwards compatibility.
 	// Setting this field is not allowed, this field is for internal use only.
 	NATOutgoingV1 bool `json:"nat-outgoing,omitempty" validate:"omitempty,mustBeFalse"`
+
+	// AllowedUse controls what the IP pool will be used for.  If not specified or empty, defaults to
+	// ["Tunnel", "Workload"] for back-compatibility
+	AllowedUses []IPPoolAllowedUse `json:"allowedUses,omitempty" validate:"omitempty"`
+
+	// AWSSubnetID if specified Calico will attempt to ensure that IPs chosen from this IP pool are routed
```

Comment:
Do we want to say explicitly that if specified, the CIDR must match the CIDR of the AWS Subnet?